articles/article-005-array-at-zero-or-one.md

```diff
@@ -0,0 +1,36 @@
+<!-- BEGIN TITLE -->
+# Should array indices start at 0 or 1?
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+
+
+> “Should array indices start at 0 or 1? 
+> My compromise of 0.5 was rejected without, I thought, proper consideration.” 
+> -Stan Kelly-Bootle
+
+An array is a data structure to store some elements with an index for each. 
+This coins the question: 'what is the smallest index?'
+
+Most programming languages start an array at index 0 because it is the smallest natural number that can be represented in a binary system. But there are some rare outliers that start with index 1. 
```

Comment:
That actually has to do with how memory is stored. The first popular programming language Fortran has an index of 1, which is not a rare outlier. C popularized the index by 0, but some mathematical based languages such as R and Matlab (again two really big languages today).